### ISSUE TITLE
Where to find file MyCaffe.basecode.dll?

### ISSUE THREAD

**Issue by zydjohnHotmail** [MyCaffe/MyCaffe#19]
Hello:
I want to see if I can run the repo.
So, I downloaded the source code, unzipped it, and open the solution by Visual Studio 2022.
But I found that there are 12 items not loadable.
When I tried to build the solution, I got 26 errors, the first one is like this:
Error	CS0006	Metadata file 'D:\GitHubRepos\MyCaffe-master\MyCaffe.basecode\bin\Debug\MyCaffe.basecode.dll' could not be found
As I can't find the missing file: MyCaffe.basecode.dll
Let me know how I can fix the issue.
By the way, if I have some pre-trained keras model (.h5) and some json format data structure of the model, can I simply load the pre-trained model and make some prediction?  If yes, I will give more details.
But, how to make the solution working?
I am using Windows 10 version (21H2), I have both VS2022 and VS2019.
Please advise!
![BuildMyCaffeNOK](https://user-images.githubusercontent.com/45263518/161383923-3c4c3a53-23d4-4b53-9af2-a7db976f6ee7.png)

**Comment by ZoroDerVonCodier** [comment]
Hi @zydjohnHotmail 

We updated the MyCaffe.sln project to only include the CUDA 11.6 builds.

To simplify things, first try unloading the CudaDnnDll.11.6, CudaPreProcDll.11.6 and MyCaffe.test.extension.11.6 and rebuild as all remaining projects are C# projects whereas the ones listed above are C++ and require CUDA - the MyCaffe.basecode.dll project should be the first to build.  

Once built, set the MyCaffe.app projects as your startup project.  You should see the MyCaffe.basecode.dll and all other MyCaffe DLL's in the following directories:

Release Build:
<clone location>\MyCaffe\MyCaffe.app\bin\Release 

Debug Build
<clone location>\MyCaffe\MyCaffe.app\bin\Debug 

Let me know if this helps!

**Comment by zydjohnHotmail** [comment]
Hello:
Thanks for your reply.  I have done the following:
1) Download the new MyCaffe.sln, unzip it and move to a folder
2) Open MyCaffe.sln with Visual Studio 2022, it shows 3 items not loaded as indicated in your reply.
3) From solution explorer, I right click on the 3 times, and delete them.
4) I run "Build Solution" from Build tools.
5) There are 26 building errors.
Please advise what I should do in this case?
![MyCaffeCUDANotLoaded](https://user-images.githubusercontent.com/45263518/161529395-3f269fee-2ee0-40ea-a5e7-fbcc3eeb0873.png)
![MyCaffeNOK](https://user-images.githubusercontent.com/45263518/161529410-930eb4b7-a53b-4c95-88a7-5495703042a6.png)

**Comment by ZoroDerVonCodier** [comment]
Hi @zydjohnHotmail ,

First, make sure you are building under Release/Debug and x64.

Ahh, I see the issue - we do not provide the *.pfx files used to sign for you must do that on your side.  To sign each project do the following:

1.) Right click the project (like MyCaffe.basecode) and select 'Properties'
2.) Select the 'Signing' tab.
3.) In the 'Choose a strong name key file:' open the drop-down and select '<New>"
4.) Add your new key file name with your password. *Make sure the file name is different from the current one listed.
5.) Repeat this step for the following projects;

MyCaffe
MyCaffe.basecode
MyCaffe.converter.onnx
MyCaffe.data
MyCaffe.db.image
MyCaffe.db.stream
MyCaffe.extras
MyCaffe.gym
MyCaffe.gym.python
MyCaffe.layers.alpha
MyCaffe.layers.beta
MyCaffe.layers.nt
MyCaffe.layers.ssd
MyCaffe.layers.hdf5
MyCaffe.model
MyCaffe.preprocessor
MyCaffe.test
MyCaffe.test.automated
MyCaffe.trainers

I just went through these steps and resolved all errors.  I will also add this to the INSTALL.md.  Let me know how this goes for you on your side.

**Comment by zydjohnHotmail** [comment]
Hello:
I download the repo again (MyCaffe-Maser.zip), unzip and copy to one folder.
I changed the CPU from anyCPU to x64. I deleted all 3 items which are not loaded.
Then, I tried to create a pfx file: MyCaffe.basecode.pfx, but it failed. See the picture.
Then, I tried to remove the item for pfx file from MyCaffe.basecode.csproj file using notepad.
  <PropertyGroup>
    <AssemblyOriginatorKeyFile>mycaffe.basecode.sn.pfx</AssemblyOriginatorKeyFile>
  </PropertyGroup>
Finally, when I tried to build the solution, I have 24 erros, a few errors less than last time.
By the way, can I use this repo to run some pre-trained model (.H5 format) to make some image prediction, like StyleGan models?
Thanks,
![MakePFXFileNOK](https://user-images.githubusercontent.com/45263518/161574554-ea38cc89-dbcc-4b2a-91a9-8dcd6dd280af.png)

**Comment by ZoroDerVonCodier** [comment]
Hi @zydjohnHotmail, 

That is strange, I just downloaded the zip, unblocked the file with powershell command "Unblock-File MyCaffe-master.zip", unzipped, opened in Visual Studio 2022 (version 17.1.3) and changed the Signing file to the same and had no problem.

Have you 'unblocked' the zip before extracting?
Also, are you using the latest version of Visual Studio 2022 (e.g. version 17.1.3)?

Perhaps, try restarting Visual Studio 2022 and try again?

You might also try some of these suggestions: https://stackoverflow.com/questions/9079466/visual-studio-popup-the-operation-could-not-be-completed

**Comment by zydjohnHotmail** [comment]
Hello:
I forgot to unblock before unzip, now I did it. I unblocked it, and unzip, and copy to one folder.
I am using Visual Studio (version 17.1.3). 
I tried to sign the basecode project, but I got the same error as last time.
I also tried to delete the pfx file from the solution, and rebuid, I got 24 errors just as before.
I also restart VS 2022 many times.
But I got the same errors as last time I showed you.
I can not find any .suo file:
C:\SoccerVideos\TensorFlow\MyCaffe-master>dir *.suo /s
 Volume in drive C has no label.
 Volume Serial Number is 8C0B-783A
File Not Found

I can't reboot now, but I will shutdown my PC tonight, and start up tomorrow.
However, I think it may be good that from your solution, do not sign the project, so I can try again without the pfx file.
The difference is that, this time, the pfx file is missing, last time, it was there.

**Comment by ZoroDerVonCodier** [comment]
Hi @zydjohnHotmail ,

I found an even easier temporary solution for you - with each of the projects, just uncheck 'Sign the assembly' and rebuild.  I have test this and it works fine on a fresh download zip of the source.  However, we would not recommend distributing signed binaries for obvious reasons.   I am not sure why you are getting the signing error in VS2022 but I would take a look at the previously mentioned article in StackOverflow.  

Alternatively, you can always use the MyCaffe NuGet download which has all signed binaries.

Hope this helps!

**Comment by zydjohnHotmail** [comment]
Hello:
Finally, I got it built successfully.  However, doing such thing is not a good experience.
But I want to know if I can load some pre-trained models (in .H5 format) to make some prediction, like the StyleGan2/3 models.
Do you have some example code some where?
thanks,

**Comment by ZoroDerVonCodier** [comment]
Hi @zydjohnHotmail ,

We will look into signing separately from each individual project and set the default to 'not sign' for I agree that may be easier.

Regarding using pre-trained models, see https://github.com/MyCaffe/MyCaffe/issues/20 which gives shows how to use the Neural Style Transfer pre-trained models.  However, if this does not hit the mark, add your question on StyleGan2/3 in that discussion thread and I will see if I can help.

**Comment by zydjohnHotmail** [comment]
OK, I will create another issue for this!
